### ISSUE TITLE
Enable SAI indexes on Cassandra 4 backends in test

### ISSUE THREAD

**Issue by olim7t** [stargate/stargate#815]
Some of our tests require SAI indexes, for example `PersistenceTest.testSecondaryIndexes()` and `RestApiv2Test.createCustomIndex()`.

In Cassandra 4, SAI must be explicitly enabled in `cassandra.yaml`:

```
enable_sasi_indexes: true
```

I've tried the following:

* set the option in `persistence-cassandra-4.0/src/test/resources/cassandra.yaml` (not sure if that one is necessary, nor if that file is used for Stargate, or the backend, or both)
* set the option in `ExternalStorage.CcmCluster`. Options are not pluggable right now, but for the sake of testing I hard-coded it in the constructor:
    ```java
     public CcmCluster(ClusterSpec spec, ExtensionContext context) {
      this(spec, ImmutableMap.of("enable_sasi_indexes", true), Collections.emptyMap(), context);
     }
    ```
* set it in `CassandraPersistenceActivator.makeConfig()` (again hard-coded):
    ```
    c.enable_sasi_indexes = true;
    ```

With a combination of all of the above, I still get an error in the backend logs when run `CassandraPersistenceIT`:

```
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.apache.cassandra.index.SecondaryIndexManager.createInstance(SecondaryIndexManager.java:768)
	at org.apache.cassandra.index.SecondaryIndexManager.createIndex(SecondaryIndexManager.java:215)
	at org.apache.cassandra.index.SecondaryIndexManager.addIndex(SecondaryIndexManager.java:278)
	at org.apache.cassandra.index.SecondaryIndexManager.reload(SecondaryIndexManager.java:200)
	at org.apache.cassandra.db.ColumnFamilyStore.reload(ColumnFamilyStore.java:259)
	at org.apache.cassandra.schema.Schema.alterTable(Schema.java:776)
	at org.apache.cassandra.schema.Schema.lambda$alterKeyspace$19(Schema.java:674)
	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
	at org.apache.cassandra.schema.Schema.alterKeyspace(Schema.java:674)
	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
	at org.apache.cassandra.schema.Schema.merge(Schema.java:656)
	at org.apache.cassandra.schema.Schema.transform(Schema.java:601)
	at org.apache.cassandra.schema.MigrationManager.lambda$announce$2(MigrationManager.java:221)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException: null
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.cassandra.index.SecondaryIndexManager.createInstance(SecondaryIndexManager.java:764)
	... 17 common frames omitted
Caused by: java.lang.NoClassDefFoundError: com/googlecode/concurrenttrees/radix/node/NodeFactory
	at org.apache.cassandra.index.sasi.memory.TrieMemIndex.<init>(TrieMemIndex.java:62)
	at org.apache.cassandra.index.sasi.memory.MemIndex.forColumn(MemIndex.java:45)
	at org.apache.cassandra.index.sasi.memory.IndexMemtable.<init>(IndexMemtable.java:42)
	at org.apache.cassandra.index.sasi.conf.ColumnIndex.<init>(ColumnIndex.java:79)
	at org.apache.cassandra.index.sasi.SASIIndex.<init>(SASIIndex.java:111)
	... 22 common frames omitted
Caused by: java.lang.ClassNotFoundException: com.googlecode.concurrenttrees.radix.node.NodeFactory
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 27 common frames omitted
```

**Comment by eribeiro** [comment]
I think we may be hitting https://issues.apache.org/jira/browse/CASSANDRA-16303 

If so, updatting C* to 4.0-beta5 would fix this particular bug (IIRC, SG is using 4.0-beta4).

**Comment by eribeiro** [comment]
According to one of C* committers, the `4.0-rc1` should be available in Maven's central repo by the end of this week. But it can be delayed due to voting process.

**Comment by eribeiro** [comment]
Hi @olim7t, C* 4.0-rc1 has finally reached Maven repo. 🎉  I am submitting a PR to your consideration. By the way, when it came to enable SASI use in C* 4.0 I followed your steps (thanks!). I wasn't to enable SASI indexes with 1), but 3) seems to do the job, I guess. Please, let me know if there's something I am missing. 🙂

**Comment by jeffreyscarpenter** [comment]
Fixed by #889
